src/clone.c

```diff
@@ -332,8 +343,9 @@ static int create_and_configure_origin(
 	git_remote_create_cb remote_create = options->remote_cb;
 	void *payload = options->remote_cb_payload;
 
-	/* If the path exists and is a dir, the url should be the absolute path */
-	if (git_fs_path_root(url) < 0 && git_fs_path_exists(url) && git_fs_path_isdir(url)) {
+	/* If the path is local and exists it should be the absolute path */
+	if (!url_has_scheme(url) && git_fs_path_root(url) < 0 &&
+	    git_fs_path_exists(url)) {
```

Comment:
I intentionally dropped `git_fs_path_isdir(url)` here since cloning from bundles could be a thing in the future.